<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>About - MGX metagenome analysis platform</title>
  <meta name="viewport" content="width=device-width, initial-scale=1">
  
  <link rel="icon" href="https://mgx-metagenomics.github.io/favicon.png">

  
  
  <link rel="stylesheet" href="/css/style.min.673d0458160597c92d4d9414ce4b7ed589d52c8c7cc87c0a988176b90e86de7f.css">
  

  

</head>

<body class='page page-default-single'>
  <div id="main-menu-mobile" class="main-menu-mobile">
  <ul>
    
    
    <li class="menu-item-home">
      <a href="/">
        <span>Home</span>
      </a>
    </li>
    
    <li class="menu-item-documentation">
      <a href="/docs/">
        <span>Documentation</span>
      </a>
    </li>
    
    <li class="menu-item-about">
      <a href="/docs/about">
        <span>About</span>
      </a>
    </li>
    
    <li class="menu-item-download">
      <a href="https://github.com/MGX-metagenomics/MGX-gui/releases/tag/v20240109">
        <span>Download</span>
      </a>
    </li>
    
  </ul>
</div>
  <div class="wrapper">
    <div class='header'>
  <div class="container">
    <div class="logo">
      <a href="https://mgx-metagenomics.github.io/"><img width="70" height="70" alt="Logo" src="/images/mgx_logo_crapola.svg" /></a>
    </div>
    <div class="logo-mobile">
      <a href="https://mgx-metagenomics.github.io/"><img  width="70" height="70" alt="Logo" src="/images/mgx_logo_crapola.svg" /></a>
    </div>
    <div id="main-menu" class="main-menu">
  <ul>
    
    
    <li class="menu-item-home">
      <a href="/">
        <span>Home</span>
      </a>
    </li>
    
    <li class="menu-item-documentation">
      <a href="/docs/">
        <span>Documentation</span>
      </a>
    </li>
    
    <li class="menu-item-about">
      <a href="/docs/about">
        <span>About</span>
      </a>
    </li>
    
    <li class="menu-item-download">
      <a href="https://github.com/MGX-metagenomics/MGX-gui/releases/tag/v20240109">
        <span>Download</span>
      </a>
    </li>
    
  </ul>
</div>
    <button id="toggle-main-menu-mobile" class="hamburger hamburger--slider" type="button">
  <span class="hamburger-box">
    <span class="hamburger-inner"></span>
  </span>
</button>
  </div>
</div>


    
    
    
    
    
    

    
    <div class="container pt-2 pt-md-6 pb-3 pb-md-6">
      <div class="row">
        <div class="col-12 col-md-3 mb-3">
          <div class="sidebar">
            
<div class="docs-menu">
  <h4>Docs</h4>
  <ul>
    
    <li class="active ">
      <a href="https://mgx-metagenomics.github.io/docs/about/">About</a>
    </li>
    
    <li class="">
      <a href="https://mgx-metagenomics.github.io/docs/install-mgx/">Install MGX</a>
    </li>
    
    <li class="">
      <a href="https://mgx-metagenomics.github.io/docs/basics/">Basics</a>
    </li>
    
    <li class="">
      <a href="https://mgx-metagenomics.github.io/docs/projects/">Projects</a>
    </li>
    
    <li class="">
      <a href="https://mgx-metagenomics.github.io/docs/analysis/">Analysis</a>
    </li>
    
    <li class="">
      <a href="https://mgx-metagenomics.github.io/docs/assembly/">Assembly</a>
    </li>
    
    <li class="">
      <a href="https://mgx-metagenomics.github.io/docs/visualization/">Visualization</a>
    </li>
    
    <li class="">
      <a href="https://mgx-metagenomics.github.io/docs/custom_pipelines/">Custom Pipelines</a>
    </li>
    
    <li class="">
      <a href="https://mgx-metagenomics.github.io/docs/references/">References</a>
    </li>
    
  </ul>
</div>

          </div>
        </div>
        <div class="col-12 col-md-9">
          
<h1 class="title">About</h1>
<div class="content ">
  <details open>
<summary><b>Table of contents</b></summary>  
<p><a href="#about">About</a><br>
<a href="#features">Features</a><br>
<a href="#citation">Citation</a><br>
<a href="#demo">Demo</a><br>
<a href="#funding">Funding</a><br>
<a href="#additional-links">Additional links</a></p>
</details>
<h2 id="about">About</h2>
<div style="text-align: justify">
<p>The characterization of microbial communities based on sequencing and analysis of their genetic information, also referred to as metagenomics, has become a popular approach; in particular, the recent advances in sequencing technologies have enabled researchers to study even the most complex communities.</p>
<p>Metagenome analysis, the assignment of sequences to taxonomic and functional entities, however, remains a tedious task: large amounts of data need to be processed. Several approaches address particular aspects, but scientific questions are often too specific to be answered by a general-purpose method.</p>
<p>We present <code>MGX</code>, a flexible and extensible <code>client/server-framework</code> for the management and analysis of metagenomic datasets. <code>MGX</code> features a comprehensive set of adaptable workflows required for taxonomic and functional metagenome analysis, combined with an intuitive and easy-to-use graphical user interface offering customizable result visualizations.</p>
<p>At the same time, <code>MGX</code> allows to include own data sources and devise custom analysis pipelines, thus enabling researchers to perform basic as well as highly specific analyses within a single application. With <code>MGX</code>, we provide a novel metagenome analysis platform that gives researchers access to the most recent analysis tools.</p>
<p><code>MGX</code> covers taxonomic and functional metagenome analysis, statistical evaluation, and a wide range of visualizations easing data interpretation. Its default taxonomic classification pipeline provides equivalent or superior results compared to existing tools.</p>
</div>
<br>
<hr>
<h2 id="features">Features</h2>
<blockquote>
<ul>
<li>Data and metadata capture</li>
<li>Quality control reports</li>
<li>analysis pipelines for taxonomic and functional analysis of raw reads</li>
<li>metagenome (co-)assembly and taxonomic binning for MAG recovery</li>
<li>de novo metatranscriptome assembly</li>
<li>workflows for taxonomic and functional annotation of assemblies</li>
<li>High-quality charts and interactive visualizations</li>
<li>Statistics: PCA, PCoA, Rarefaction, ..</li>
<li>Fragment recruitment</li>
<li>Sequence export</li>
<li>Custom pipeline support</li>
</ul>
</blockquote>
<hr>
<h2 id="citation">Citation</h2>
<p><em>MGX 2.0: Shotgun- and assembly-based metagenome and metatranscriptome analysis from a single source</em>
Sebastian Jaenicke, Sonja Diedrich, and Alexander Goesmann
bioRxiv 2023.09.21.558800
DOI: <a href="https://www.biorxiv.org/content/10.1101/2023.09.21.558800v1">10.1101/2023.09.21.558800</a></p>
<p><em>Flexible metagenome analysis using the MGX framework</em>
Sebastian Jaenicke, Stefan P. Albaum, Patrick Blumenkamp, Burkhard Linke, Jens Stoye and Alexander Goesmann
Microbiome 2018 6:76
DOI: <a href="https://doi.org/10.1186/s40168-018-0460-1">10.1186/s40168-018-0460-1</a></p>
<hr>
<h2 id="demo">Demo</h2>
<p>If you want to try out MGX 2.0, we offer two public demonstration projects:</p>
<blockquote>
<p><code>MGX2_Demo</code> contains sequence data from the black band disease study previously <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5240343/">published</a> by Sato et al. that was used as a demonstration example in the upcoming MGX 2.0 publication; ENA project accession <a href="https://www.ebi.ac.uk/ena/browser/view/PRJNA321354">PRJNA321354</a>.</p>
</blockquote>
<blockquote>
<p><code>MGX2_TermiteGut</code> with metagenome and metatranscriptome date from the well-known termite gut study; ENA project accession <a href="https://www.ebi.ac.uk/ena/browser/view/PRJNA454696">PRJNA454696</a>.</p>
</blockquote>
<p>To access these example projects, download the MGX 2.0 application and log in with username <code>mgxdemo</code> and password <code>mgxdemo</code>. Please note that only read access is permitted here, <em>i.e.</em> you can explore all precomputed results and inspect the assemblies, but wont be able to initiate new analyses or delete existing data.</p>
<hr>
<h2 id="funding">Funding</h2>
<p><a href="/images/denbi_logo.png"><img src="/images/denbi_logo.png"  width="200"/></a></p>
<p>Funding for the operation and maintenance of MGX is provided by the <a href="https://www.bmbf.de/">German Federal Ministry of Education and Research (BMBF)</a> project <a href="https://www.denbi.de/bigi">Bielefeld-Gie√üen Center for Microbial Bioinformatics - BiGi</a> (grant 031A533) within the German Network for Bioinformatics Infrastructure <a href="https://www.denbi.de/">de.NBI</a>.</p>
<hr>
<h2 id="additional-links">Additional links</h2>
<ul>
<li>Synthetic metagenomes used in the MGX manuscript are available from <a href="https://doi.org/10.5281/zenodo.4382271">Zenodo</a></li>
<li><a href="https://www.computational.bio.uni-giessen.de/">Bioinformatics and Systems Biology at JLU Giessen</a></li>
<li><a href="https://www.uni-giessen.de/fbz/fb08/Inst/bioinformatik/bcf/data-privacy">Data privacy statement</a></li>
<li><a href="https://www.denbi.de/">de.NBI - German network for bioinformatics infrastructure</a></li>
<li><a href="https://www.kegg.jp/feedback/copyright.html">KEGG copyright request</a> <em>needed if you want to publish a KEGG pathway map created within MGX</em></li>
</ul>
<p><img src="/images/slides/3.png" alt="Not so big"></p>

</div>
</div>

        </div>
      </div>
    </div>
    
  </div>

  <div class="sub-footer">
  <div class="container">
    <div class="row">
      <div class="col-12">
        <div class="sub-footer-inner">
          
            
<div class="social">
    
        <a href="https://github.com/MGX-metagenomics" target="blank"><img height="20" width="20" src="/images/social/github.svg" title="Github" alt="Github" /></a>
    
        <a href="https://twitter.com/zerostaticio" target="blank"><img height="20" width="20" src="/images/social/twitter.svg" title="Twitter" alt="Twitter" /></a>
    
</div>

          
          
            <div class="copyright"><a class="MGX" href="https://github.com/MGX-metagenomics">MGX</a></div>
          
        </div>
      </div>
    </div>
  </div>
</div>


  

  
  

  
  <script type="text/javascript" src="/js/scripts.min.302a6fc365d5139fb98cf60bdb8f715d96257ea189161d36c190ccfa8182e569.js"></script>
  

  
  
  
    
  


</body>

</html>
